include/nanorange/algorithm/stable_partition.hpp

```diff
@@ -95,50 +97,52 @@ struct stable_partition_fn {
         }
 
         const I first_false = (m1 == first) ? first :
-                impl_unbuffered(first, m1, len_half, pred, proj);
+                impl_unbuffered(first, m1, len_half, pred, proj).begin();
 
         m1 = middle;
         len_half = dist - half;
 
         while (nano::invoke(pred, nano::invoke(proj, *m1))) {
             if (++m1 == last) {
-                return nano::rotate(first_false, middle, ++last).begin();
+                auto rot = nano::rotate(first_false, middle, ++last);
+                return {std::move(rot.begin()), std::move(last)};
             }
         }
 
-        const I last_false = impl_unbuffered(m1, last, len_half, pred, proj);
+        const I last_false = impl_unbuffered(m1, last, len_half, pred, proj).begin();
```

Comment:
After correcting the return values (from returning the "left" partition to returning the "right" partition), I accidentally forgot to change this from `end()` to `begin()` and it wasn't caught by the tests.

Same bug slipped a few lines above.